packages/zowe-explorer-api/src/vscode/ui/utils/TableBuilder.ts

```diff
@@ -0,0 +1,224 @@
+    public columns(...columns: Table.ColumnOpts[]): this {
+        this.data.columns = this.convertColumnOpts(columns);
+        return this;
+    }
+
+    private convertColumnOpts(columns: Table.ColumnOpts[]): Table.Column[] {
+        return columns.map((col) => ({
+            ...col,
+            comparator: col.comparator?.toString(),
+            colSpan: col.colSpan?.toString(),
+            rowSpan: col.rowSpan?.toString(),
+            valueFormatter: col.valueFormatter?.toString(),
+        }));
+    }
+
+    /**
+     * Adds columns to the table. Does not replace existing columns.
+     * @param columns The column definitions to add to the table
+     * @returns The same {@link TableBuilder} instance with the new column definitions added
+     */
+    public addColumns(columns: Table.ColumnOpts[]): this {
+        this.data.columns = [...this.data.columns, ...this.convertColumnOpts(columns)];
+        return this;
+    }
+
+    /**
+     * Add context options for the next table.
+     * @param actions the record of indices to {@link Table.Action} arrays to use for the table
+     * @returns The same {@link TableBuilder} instance with the row actions added
+     */
+    public contextOptions(opts: Record<number | "all", Table.ContextMenuOpts[]>): this {
+        for (const [key, optsForKey] of Object.entries(opts)) {
+            for (const opt of optsForKey) {
+                this.addContextOption(key as number | "all", opt);
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Add a context menu option to the table.
+     * @param index The row index to add an option to (or "all" for all rows)
+     * @returns The same {@link TableBuilder} instance with the context menu option added
+     */
+    public addContextOption(index: number | "all", option: Table.ContextMenuOpts): this {
+        if (this.data.contextOpts[index]) {
+            const opts = this.data.contextOpts[index];
+            this.data.contextOpts[index] = [...opts, { ...option, condition: option.condition?.toString() }];
+        } else {
+            this.data.contextOpts[index] = [{ ...option, condition: option.condition?.toString() }];
+        }
+        return this;
+    }
+
+    /**
+     * Add row actions for the next table.
+     * @param actions the record of indices to {@link Table.Action} arrays to use for the table
+     * @returns The same {@link TableBuilder} instance with the row actions added
+     */
+    public rowActions(actions: Record<number | "all", Table.ActionOpts[]>): this {
+        for (const key of Object.keys(actions)) {
+            this.addRowAction(key as number | "all", actions[key]);
+        }
+        return this;
+    }
+
+    /**
+     * Add a row action to the next table.
+     * @param index The column index to add an action to
+     * @returns The same {@link TableBuilder} instance with the row action added
+     */
+    public addRowAction(index: number | "all", action: Table.ActionOpts): this {
```

Comment:
I like the thought of passing in `all`

was this supposed to allow for all rows to have the same action?
If so, should we be checking for it?

I might be missing something here. 🤔

Reply:
Good question - discussed offline and we do support the "all" keyword as an "index" into the record. It is strange but technically saves memory and keeps all the actions in one contiguous structure 😋